rust/lance/src/index/vector.rs

```diff
@@ -439,5 +439,10 @@ pub(crate) async fn open_vector_index_v2(
         }
     };
 
+    dataset
+        .session
+        .index_cache
+        .insert_vector(uuid, index.clone());
+
```

Comment:
Where's the corresponding "cache read"?  Normally I'd expect to see something like...

```
if let Some(index) = cache.get(key) {
  index
} else {
  let index = create_thing(...);
  cache.insert(index);
  index
}
```

Reply:
yep this is the confusing part of this code. The cache is read from a different function `open_vector_index`. I'll add a ticket to clean this up. This is just keep it on par with v1